### PR TITLE
[snapcraft] use buildd images for core18 and core20

### PR DESCRIPTION
**Pull request by Saviq**

### CODE REVIEW

File: src/daemon/custom_image_host.cpp [hunk 1]
```diff
@@ -68,16 +68,16 @@ const QMap<QString, CustomImageInfo> multipass_image_info{
 const QMap<QString, CustomImageInfo> snapcraft_image_info{
     {{"ubuntu-16.04-minimal-cloudimg-amd64-disk1.img"},
      {"https://cloud-images.ubuntu.com/minimal/releases/xenial/release/",
-      {"core", "core16"},
```
Comment (sergiusens) [inline src/daemon/custom_image_host.cpp:71]: Please do not remove core, it is a valid use case:

- as a trigger for Snapcraft 3.0 functionality
- as the default when using multipass without a base (e.g.; OS X).

It is also documented here https://snapcraft.io/docs/base-snaps

**Review by sergiusens — CHANGES_REQUESTED** [review]
multipass snap works fine, when using base core20 and core18, it does not though for base core.

This:

```yaml
name: play-core
base: core
version: '0.1'
summary: test using core as base
description: |
  filler...
  ...

grade: devel
confinement: devmode

parts:
  my-part:
    plugin: nil
```

results in:

```
$ snapcraft
Launching a VM.
launch failed: Unable to find an image matching "core"                          
An error occurred with the instance when trying to launch with 'multipass': returned exit code 2.
Ensure that 'multipass' is setup correctly and try again.
```

**Comment by sergiusens** [comment]
Considering the linked documentation and the fact that core16 is not ready, perhaps drop that instead.

File: src/daemon/custom_image_host.cpp (hunk 1 above)
Comment (Saviq) [reply src/daemon/custom_image_host.cpp:71]: Of course. I confused the two… guess why :P

Fixed.

**Comment by Saviq** [comment]
@sergiusens I took your YAML and ran this on qemu (they don't boot yet on LXD):

```bash
$ bash -euo pipefail -c '
for base in core{20,18,}; do
  sed -i "s/^base:.*/base: $base/" snapcraft.yaml
  snapcraft
  multipass delete --purge --all
done'
```

The same for `core{18,20}` on macOS on hyperkit and VirtualBox. Also launched those two on Windows on Hyper-V and VirtualBox.

All good from my POV.

File: src/platform/platform_linux.cpp [hunk 2]
```diff
@@ -154,6 +154,12 @@ bool mp::platform::link(const char* target, const char* link)
 
 bool mp::platform::is_alias_supported(const std::string& alias, const std::string& remote)
 {
+    auto driver = utils::get_driver_str();
+
+    // snapcraft:core{18,20} images don't work on LXD yet
```
Comment (townsend2010) [inline src/platform/platform_linux.cpp:159]: What is the status of these images working with LXD and why don't they work?

Comment (Saviq) [reply src/platform/platform_linux.cpp:159]: They don't have working UEFI booting. It's on their radar. I know this somewhat regresses the LXD backend, but the alternative is to keep `minimal` images for LXD, and use `buildd` for others… I'd rather not.

I was also wondering whether we should disallow the `snapcraft:` remote on LXD altogether.

Comment (townsend2010) [reply src/platform/platform_linux.cpp:159]: Ok, this was all taken care of on chat.

**Review by townsend2010 — CHANGES_REQUESTED** [review]
I think this is good once LXD support for the `snapcraft:` remote is disabled.

**Review by townsend2010 — CHANGES_REQUESTED** [review]
Test coverage dropped in the last commit.  Probably need to add back in something like https://github.com/canonical/multipass/pull/1679/files#diff-7ef9608cecf497ebab4b6ca882daa3f62d9438804c5072ee37257e08806acf54.

**Comment by Saviq** [comment]
> Test coverage dropped in the last commit.

Fixed.

**Review by townsend2010 — APPROVED** [review]
LGTM

bors merge

**Pull request merged**